### PR TITLE
Adds a user manual to dpctl documentation

### PR DESCRIPTION
**Pull request by diptorupd**
The PR will add a new user manual demonstrating the various features of `dpctl` from an end-user perspective.

### CODE REVIEW

**Comment by diptorupd** [comment]
@oleksandr-pavlyk I closed the previous PR and opened the fresh one from a branch on our main repo rather than my fork.

File: docs/docfiles/user_guides/manual/intro.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+and extension developers to access the data-parallel computing resource
+available on a modern heterogeneous system. Dpctl's objective is to provide
+a runtime to control or to manage data-parallel devices without the need for
+other Python packages to create such a runtime themselves. The runtime is built
+on top of the C++ SYCL standard and is envisioned to be both vendor and
+architecture agnostic. If the underlying SYCL runtime supports a type of
+architecture, the dpctl runtime will allow managing that architecture from
+Python.
+
+In its current implementation, dpctl supports only the DPC++ SYCL
```
Comment (oleksandr-pavlyk) [inline docs/docfiles/user_guides/manual/intro.rst:16]: In its current form, `dpctl` relies on certain DPC++ extensions of SYCL standard. Moreover, the binary distribution of `dpctl` uses the proprietary Intel(R) oneAPI DPC++ runtime  bundled as part of oneAPI and supports Intel XPU devices only. However, `dpctl` is compatible with the runtime of open-source DPC++ sycl bundle that can be compiled to support a wide range of architectures (or backends?) including CUDA, AMD ROC, and HIP.

Comment (diptorupd) [reply docs/docfiles/user_guides/manual/intro.rst]: Thanks!

File: docs/docfiles/user_guides/manual/intro.rst [hunk 2]
```diff
@@ -29,28 +29,45 @@ and native extension developers should refer to the programmer's guide.
...
 
-Data parallelism enables access to parallel resources in a modern
-heterogeous system.
+* **Platform**
+    A device driver installed on the system is termed as a platform. As multiple
+    devices of the same type can share the same device driver, a platform may
+    contain multiple devices. Note that the same physical hardware (say, a GPU)
+    may be reflected as two separate devices if they can be programmed by more
+    than one platform, *e.g.*, the same GPU hardware can be listed as an
+    OenCL GPU device and a Level-Zero GPU device.
```
Comment (oleksandr-pavlyk) [inline docs/docfiles/user_guides/manual/intro.rst:57]:
```suggestion
    OpenCL GPU device and a Level-Zero GPU device.
```

File: docs/docfiles/user_guides/manual/dpctl/intro.rst [hunk 3]
```diff
@@ -3,17 +3,17 @@
...
+The Data Parallel Control (dpctl) package provides a Python runtime to access a
+data-parallel computing resource or *XPU* from another Python application or
+library, alleviating the need for the other Python packages to develop such a
+runtime themselves. The term XPU denotes a diverse range of compute
+architectures such as a CPU, GPU, FPGA, *etc.*, available to programmers on a
+modern heterogeneous system.
+
+The dpctl runtime is built on top of the C++ SYCL standard and is designed to be
+both vendor and architecture agnostic. If the underlying SYCL runtime supports
+a type of architecture, the dpctl runtime will allow managing that architecture
```
Comment (oleksandr-pavlyk) [inline docs/docfiles/user_guides/manual/dpctl/intro.rst:15]:
```suggestion
a type of architecture, the dpctl runtime will allow accessing that architecture
```

**Comment by diptorupd** [comment]
@oleksandr-pavlyk A rough draft for device selection is now added.

File: docs/docfiles/user_guides/manual/dpctl/device_selection.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+.. program-output:: python ../examples/python/device_selection.py -r create_gpu_device_if_present
+
+.. Note::
+    A **filter string** is a three-tuple that may specify the *backend*,
+    *device type*, and *device number* as a colon (:) separated string. The
+    backend specifies the type of device driver and can have a value such as
+    *opencl*, *level-zero*, or *cuda*. The device type can be *gpu*, *cpu*,
+    *accelerator*. And, the device number is a numeric value specifying the
+    ordinality of the device in the listing of devices as determined by the SYCL
+    runtime. The filter values are dependent of what the underlying SYCL runtime
```
Comment (oleksandr-pavlyk) [inline docs/docfiles/user_guides/manual/dpctl/device_selection.rst:91]: We should state that the listing of devices given by the SYCL runtime is stable (same one seen by different processes, or from run to run, provided that driver configuration and `SYCL_DEVICE_FILTER` setting are the same.) In other words, `dpctl.SyclDevice("4")` will select the same device as using `sycl::ext::oneapi::filter_selector("4")` in a C++ application.

File: docs/docfiles/user_guides/manual/dpctl/device_selection.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+.. literalinclude:: ../../../../../examples/python/device_selection.py
+    :language: python
+    :lines: 20-21, 70-91
+    :caption: Custom Device Selection
+    :linenos:
+
+The example :ref:`fig-custom-device-selection` shows a way of selecting a device
+based off a specific hardware property. The :func:`dpctl.get_devices()` returns
+a list of all devices on the system, out of that list the devices that support
```
Comment (oleksandr-pavlyk) [inline docs/docfiles/user_guides/manual/dpctl/device_selection.rst:118]:
```suggestion
a list of all unpartitioned devices on the system, out of that list the devices that support
```

Comment (diptorupd) [reply docs/docfiles/user_guides/manual/dpctl/device_selection.rst]: Done. Please review once again.

File: docs/docfiles/user_guides/manual/dpctl/device_selection.rst (hunk 4 above)
Comment (diptorupd) [reply docs/docfiles/user_guides/manual/dpctl/device_selection.rst]: Done. Please review again.

**Comment by diptorupd** [comment]
@oleksandr-pavlyk I cleaned up and rebased the commits. Review once and we can merge.

**Review by oleksandr-pavlyk — APPROVED** [review]
Thank you @diptorupd

**Pull request merged**